src/test/java/org/prebid/server/bidder/ix/IxBidderTest.java

```diff
@@ -636,6 +636,154 @@ public void makeBidsShouldReturnValidAdmIfNativeIsPresentInImpAndAdm12() throws
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue())
+                .extracting(BidderBid::getType)
+                .containsExactly(BidType.audio);
+    }
+
+    @Test
+    public void makeBidsShouldReturnNativeBidIfMTypeIsFour() throws JsonProcessingException {
+        // given
+        final Banner banner = Banner.builder().w(300).h(200).build();
+        final Video video = Video.builder().build();
+        final BidderCall<BidRequest> httpCall = givenHttpCall(
+                BidRequest.builder()
+                        .imp(singletonList(Imp.builder()
+                                .id("123")
+                                .banner(banner)
+                                .video(video).build()))
+                        .build(),
+                mapper.writeValueAsString(
+                        givenBidResponse(
+                                bidBuilder -> bidBuilder
+                                        .impid("123")
+                                        .mtype(4))));
+
+        // when
+        final Result<List<BidderBid>> result = ixBidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue())
+                .extracting(BidderBid::getType)
+                .containsExactly(BidType.xNative);
+    }
+
+    @Test
+    public void makeBidsShouldReturnCorrectTypeExtPrebidTypeInResponse() throws JsonProcessingException {
+        // given
+        final Banner banner = Banner.builder().w(300).h(200).build();
+        final Video video = Video.builder().build();
+        final BidderCall<BidRequest> httpCall = givenHttpCall(
+                BidRequest.builder()
+                        .imp(singletonList(Imp.builder()
+                                .id("123")
+                                .banner(banner)
+                                .video(video).build()))
+                        .build(),
+                mapper.writeValueAsString(
+                        givenBidResponse(
+                                bidBuilder -> bidBuilder
+                                        .impid("123")
+                                        .ext(mapper.createObjectNode()
+                                                .set("prebid", mapper.createObjectNode().put("type", "video"))))));
+
+        // when
+        final Result<List<BidderBid>> result = ixBidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue())
+                .extracting(BidderBid::getBid)
+                .extracting(Bid::getExt)
+                .extracting(ext -> ext.get("prebid"))
+                .extracting(node -> mapper.treeToValue(node, ExtBidPrebid.class))
+                .extracting(ExtBidPrebid::getType)
+                .containsExactly(BidType.video);
```

Comment:
Why do we need to assert our input?

Reply:
Only the adapter can affect this result. If logic appears that somehow changes this object, the corresponding test will be written.